backend/src/test/java/com/redhat/cloud/notifications/recipients/rbac/AuthRequestFilterTest.java

```diff
@@ -0,0 +1,75 @@
+package com.redhat.cloud.notifications.recipients.rbac;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.List;
+
+public class AuthRequestFilterTest {
+
+    @BeforeEach
+    public void clean() {
+        System.setProperty("rbac.service-to-service.exceptional.auth.info", "");
+    }
+
+    @Test
+    public void testServiceToServiceHeaders() throws IOException {
+        ClientRequestContext context = configureContext();
+
+        AuthRequestFilter rbacAuthRequestFilter = new AuthRequestFilter();
+        rbacAuthRequestFilter.application = "My nice app";
+        rbacAuthRequestFilter.secret = "this-is-a-secret-token";
+
+        rbacAuthRequestFilter.filter(context);
+        MultivaluedMap<String, Object> map = context.getHeaders();
+        Assertions.assertEquals("this-is-a-secret-token", context.getHeaderString("x-rh-rbac-psk"));
+        Assertions.assertEquals("My nice app", context.getHeaderString("x-rh-rbac-client-id"));
+        Assertions.assertNull(context.getHeaderString("Authorization"));
+    }
+
+    @Test
+    public void testDevServiceToServiceHeaders() throws IOException {
+        System.setProperty("rbac.service-to-service.exceptional.auth.info", "myuser:p4ssw0rd");
+        ClientRequestContext context = configureContext();
+
+        AuthRequestFilter rbacAuthRequestFilter = new AuthRequestFilter();
+        rbacAuthRequestFilter.application = "My nice app";
+        rbacAuthRequestFilter.secret = "this-is-a-secret-token";
+
+        // Setting x-rh-rbac-account
+        context.getHeaders().putSingle("x-rh-rbac-account", "the-account-id");
+
+        rbacAuthRequestFilter.filter(context);
+        Assertions.assertNull(context.getHeaderString("x-rh-rbac-psk"));
+        Assertions.assertNull(context.getHeaderString("x-rh-rbac-client-id"));
+
+        // Account is removed
+        Assertions.assertNull(context.getHeaderString("x-rh-rbac-account"));
+        Assertions.assertEquals(
+                "Basic " + Base64.getEncoder().encodeToString("myuser:p4ssw0rd".getBytes()),
```

Comment:
This is harmless here but it could be copied someday into runtime code and become a problem:
```suggestion
                "Basic " + new String(Base64.getEncoder().encode("myuser:p4ssw0rd".getBytes(UTF_8)), UTF_8),
```
Requires: `import static java.nio.charset.StandardCharsets.UTF_8;`